app/backend/prepdocslib/cloudingestionstrategy.py

```diff
@@ -0,0 +1,334 @@
+            auth_identity=SearchIndexerDataUserAssignedIdentity(
+                resource_id=self.search_user_assigned_identity_resource_id
+            ),
+            inputs=[
+                InputFieldMappingEntry(name="consolidated_document", source="/document/consolidated_document"),
+            ],
+            outputs=[OutputFieldMappingEntry(name="chunks", target_name="chunks")],
+        )
+
+        return SearchIndexerSkillset(
+            name=self.skillset_name,
+            description="Skillset linking document extraction, figure enrichment, and text processing functions",
+            skills=[document_extractor_skill, figure_processor_skill, shaper_skill, text_processor_skill],
+            index_projection=index_projection,
+        )
+
+    async def setup(self) -> None:
+        logger.info("Setting up search index and skillset for cloud ingestion")
+
+        if not self.embeddings.azure_endpoint or not self.embeddings.azure_deployment_name:
+            raise ValueError("Integrated vectorization requires Azure OpenAI endpoint and deployment")
+
+        if not isinstance(self.embeddings, OpenAIEmbeddings):
+            raise TypeError("Cloud ingestion requires Azure OpenAI embeddings to configure the search index.")
+
+        self._search_manager = SearchManager(
+            search_info=self.search_info,
+            search_analyzer_name=self.search_analyzer_name,
+            use_acls=self.use_acls,
+            use_parent_index_projection=True,
+            embeddings=self.embeddings,
+            field_name_embedding=self.search_field_name_embedding,
+            search_images=self.use_multimodal,
+            enforce_access_control=self.enforce_access_control,
+        )
+
+        await self._search_manager.create_index()
+
+        async with self.search_info.create_search_indexer_client() as indexer_client:
+            data_source_connection = SearchIndexerDataSourceConnection(
+                name=self.data_source_name,
+                type=SearchIndexerDataSourceType.AZURE_BLOB,
+                connection_string=self.blob_manager.get_managedidentity_connectionstring(),
+                container=SearchIndexerDataContainer(name=self.blob_manager.container),
+                data_deletion_detection_policy=NativeBlobSoftDeleteDeletionDetectionPolicy(),
+            )
+            await indexer_client.create_or_update_data_source_connection(data_source_connection)
+
+            skillset = self._build_skillset()
+            await indexer_client.create_or_update_skillset(skillset)
+
+    async def run(self) -> None:
+        if self.document_action == DocumentAction.Add:
+            files = self.list_file_strategy.list()
+            async for file in files:
+                try:
+                    await self.blob_manager.upload_blob(file)
+                finally:
+                    if file:
+                        file.close()
+        elif self.document_action == DocumentAction.Remove:
```

Comment:
We support native soft delete, customer should manage blob container directly

Reply:
Removed Remove/RemoveAll here